### ISSUE TITLE
Bug: [space-infix-ops] Spurious warning after upgrade to 5.27.0

### ISSUE THREAD

**Issue by doberkofler** [typescript-eslint/typescript-eslint#5109]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=MYGwhgzhAEAqBOYB2FwBcCmswCMQegG8AoaaAB3gEsA3MTaAE3rAC4joB7agc3YjTUkPADTQkYALYZ+gqsOgBfANoBdaAF5oagNzFFxIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZYgQzMVuYDNKAD1oB7YqgyQA7p2hNI4MAF8QyoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class TranslateTable {
  private data: { orig: string, name: string }[] = [];
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/space-infix-ops": "warn",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No warning as in 5.26.0

### Actual Result

Warning

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.27.0` |
| `@typescript-eslint/parser`            | `5.27.0` |
| `TypeScript`                           | `4.7.2` |
| `ESLint`                               | `8.16.0` |
| `node`                                 | `18.1.0` |
